modules/queue/base_redis_test.go

```diff
@@ -56,7 +56,7 @@ func TestBaseRedis(t *testing.T) {
 	}()
 	if !waitRedisReady("redis://127.0.0.1:6379/0", 0) {
 		redisServer = redisServerCmd(t)
-		if redisServer == nil && os.Getenv("CI") != "" {
+		if redisServer == nil && os.Getenv("CI") == "" {
```

Comment:
How about just test if redisServer is nil? Whatever is it in CI or not, the warnings are both meaningful.
```suggestion
		if redisServer == nil {
```

Reply:
I don't think we should pass CI while silently skipping test. For instance, if our redis service stops working we will never know. 

See #24650

Reply:
> I don't think we should pass CI while silently skipping test. For instance, if our redis service stops working we will never know.
> 
> See #24650

I think yours is better.